spring-cloud-dataflow-core/src/test/java/org/springframework/cloud/dataflow/core/dsl/StreamParserTests.java

```diff
@@ -339,73 +340,53 @@ public void expressions_gh1_2() {
 
 	@Test
 	public void sourceChannel() {
-		StreamNode sn = parse("queue:foobar > file");
-		assertEquals("[(queue:foobar:0>12)>(ModuleNode:file:15>19)]", sn.stringify(true));
+		StreamNode sn = parse(":foobar > file");
+		assertEquals("[(foobar:1>7)>(ModuleNode:file:10>14)]", sn.stringify(true));
 	}
 
 	@Test
 	public void sinkChannel() {
-		StreamNode sn = parse("http > queue:foo");
-		assertEquals("[(ModuleNode:http:0>4)>(queue:foo:7>16)]", sn.stringify(true));
+		StreamNode sn = parse("http > :foo");
+		assertEquals("[(ModuleNode:http:0>4)>(foo:8>11)]", sn.stringify(true));
 	}
 
 	@Test
 	public void channelVariants() {
-		// Job is not a legal channel prefix
-		checkForParseError("trigger > job:foo", DSLMessage.EXPECTED_CHANNEL_PREFIX_QUEUE_TOPIC, 10, "job");
 
 		// This looks like a label and so file is treated as a sink!
-		checkForParseError("queue: bar > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 7);
+		checkForParseError("http > :test value", DSLMessage.UNEXPECTED_DATA_AFTER_STREAMDEF, 13);
 
-		// 'queue' looks like a module all by itself so everything after is unexpected
-		checkForParseError("queue : bar > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 6);
+		checkForParseError(":boo .xx > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 5);
+		checkForParseError(":boo . xx > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 5);
+		checkForParseError(":boo. xx > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 6);
+		checkForParseError(":boo.xx. yy > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 9);
+		checkForParseError(":boo.xx .yy > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 8);
+		checkForParseError(":boo.xx . yy > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 8);
```

Comment:
Here the `colon` is parsed as a token and the DSL message `NO_WHITESPACE_IN_CHANNEL_DEFINITION` is thrown only if the parser finds a whitespace `around` the DOT inside the `identifier` that is parsed after the `colon`. Hence, the whitespace after the `:` is ignored here. I believe this is ok given it is only at the beginning after colon.